### PR TITLE
Improving vignette/plots

### PR DESCRIPTION
**Pull request by hadley**
Apologies for the state of this PR - I started worked on the vignette, discovered that some plots didn't work, and then in did a bunch of refactoring.

By and large the plots should look the same as before, although I've considerable simplified the underlying code and made a few minor tweaks here and there. The major difference is that sessions in warm-up/cool-down are no longer coloured differently. There's no straightforward way to do this, and I think it only worked previously due to a bug in manual scales.

I still need to finish the vignette, but I wanted to capture my progress so far.

### CODE REVIEW

**Comment by hadley** [comment]
Ok, this is ready to review. In brief, I've:

* Overhauled the homepage, so you get to the meat of the package more quickly.
* Created a new shinycannon article that gives the details that were previously on the homepage.
* Rewritten the analysis vignette to get into the plots more quickly, and to give more advice on interpretation.
* Tweaked the `_pkgdown.yml` so that the navbar uses more data from the vignettes
* Refactored the plotting code to reduce duplication
* Fixed the bug that lead to many plots being blank. I believe this code only worked before because of a bug in ggplot2 that's now been fixed. I've now simply removed the sessions from the warmup/cooldown phases in those plots.
* Made a few tweaks to the plots as described in `NEWS.md`

**Comment by hadley** [comment]
I'd recommend reviewing by building the site locally so that you can read the updated vignette to see the tweaked plots and read the accompanying explanation.

File: R/analysis.R [hunk 1]
```diff
@@ -10,70 +10,38 @@ strip_suffix <- function(str) {
...
+    col_types = vroom::cols(
+      session_id = vroom::col_integer(),
+      worker_id = vroom::col_integer(),
+      iteration = vroom::col_integer(),
+      event = vroom::col_character(),
+      timestamp = vroom::col_double(),
+      input_line_number = vroom::col_integer(),
+      comment = vroom::col_character()
+    ),
+    comment = "#",
```
Comment (schloerke) [inline R/analysis.R:30]:
```suggestion
    comment = "#",
    progress = verbose
```

**Comment by schloerke** [comment]
Thank you for the awesome updates, @hadley !!

**Pull request merged**